idl/cdc_topology_description.idl.hh

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2019 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace cdc {
+
+class stream_id final {
+    int64_t _first;
+    int64_t _second;
+};
+
+class topology_description final {
+    std::vector<dht::token> _tokens;
+    std::vector<std::vector<cdc::stream_id>> _streams;
```

Comment:
Should explain what the first and second index of the nested vector are.

In case the first index is the same as that for tokens, better to use array-of-struct than struct-of-arrays:

```c++
class topology_description_token_entry {
    dht::token _token;
    std::vector<cdc::stream_id> _streams; // comment here explaining what this means
    uint8_t _ignore_msb_bits;
};

Reply:
cdc/generation.hh contains explanation of topology_description. I guess I'll add a reference.

Reply:
I switched to array-of-struct and added a reference to cdc/generation.hh